<?xml version="1.0"?>
<launch>
	<arg name="cf_leader_name" value="demo_crazyflie10" />
	<!-- initial reference -->
	<arg name="cf_1_x" value="0" /> 
	<arg name="cf_1_y" value="4" />
	<arg name="cf_1_z" value="0.5" />
	
	<param name="cf_leader_x" type="double" value="$(arg cf_1_x)" />
	<param name="cf_leader_y" type="double" value="$(arg cf_1_y)" />
	<param name="cf_leader_z" type="double" value="$(arg cf_1_z)" />


	<arg name="cf_follower_name" value="demo_crazyflie9" />
	<!-- initial reference -->
  	<arg name="cf_2_x" value="-1" />
	<arg name="cf_2_y" value=" 4" />
	<arg name="cf_2_z" value=" 0.5" />

	<!-- Initialize global parameters -->
	<param name="cf_leader_name" type="string" value="$(arg cf_leader_name)" />
	<param name="cf_follower_name" type="string" value="$(arg cf_follower_name)" />

	<param name="cf_follower_x" type="double" value="$(arg cf_2_x)" />
	<param name="cf_follower_y" type="double" value="$(arg cf_2_y)" />
	<param name="cf_follower_z" type="double" value="$(arg cf_2_z)" />
	
	
	

	<group ns="cf_leader">
		<!-- <node name="cf_controller" pkg="execution" type="hummingbird3.py"    args="$(arg cf_leader_name)  -->
			<!-- $(arg cf_1_x) $(arg cf_1_y) $(arg cf_1_z)" /> -->
  		<!-- <node name="traj_executor" pkg="execution" type="traj_executor_position_controller.py" args="$(arg cf_leader_name)" /> -->
  		<!-- <node name="traj_executor" pkg="execution" type="leader_traj_executor_position_controller.py" output="screen" args="$(arg cf_leader_name)" /> -->
	</group>

	<group ns="cf_follower">
		<!-- <node name="cf_controller" 	pkg="execution" type="hummingbird3.py"  output="screen" args="$(arg cf_follower_name)   -->
			<!-- $(arg cf_2_x) $(arg cf_2_y) $(arg cf_2_z) " /> -->
  		<!-- <node name="traj_executor" pkg="execution" type="traj_executor_position_controller.py" output="screen" args="$(arg cf_follower_name)" /> -->
  		<!-- <node name="traj_executor" pkg="execution" type="follower_traj_executor_position_controller.py" output="screen" args="$(arg cf_follower_name) $(arg cf_leader_name) " /> -->
	
	</group>

	
	<!-- Visualizations  -->
	<!-- <arg name="env_mesh_path" value="drone_path_planning/resources/stl/env-scene-ltu-experiment.stl" /> -->
	<!-- <arg name="env_mesh_path" value="drone_path_planning/resources/collada/env-scene-ltu-experiment-hole-inclined.dae" /> -->
	<!-- <arg name="env_mesh_path" value="drone_path_planning/resources/collada/env-scene-ltu-experiment-corridor-narrow.dae" />	 -->
	<!-- <arg name="env_mesh_path" value="drone_path_planning/resources/collada/env-scene-ltu-experiment-corridor-narrow-easy.dae" />	 -->
	<!-- <arg name="env_mesh_path" value="drone_path_planning/resources/collada/env-scene-ltu-experiment-small-hole.dae" /> -->
	<arg name="env_mesh_path" value="drone_path_planning/resources/collada/env-scene-ltu-experiment-tunnel2.dae" />	
	
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find execution)/launch/formation.rviz" output="log"/>
	<node name="mesh_renderer" pkg="3Dvisualization" type="render_mesh.py"  args="$(arg env_mesh_path)" />
	<node name="drones_vis"    pkg="3Dvisualization" type="droneMarkerVis.py" />
	<node name="catenaries_listener"    pkg="catenaries" type="catenaries_listener.py"  args="1"  />  	<!-- TODO:add rope length as param -->
	<!-- <node name="error_calc"    pkg="3Dvisualization" type="error_calc.py"  /> -->

	<!-- <node name="traj_executor1" pkg="crazyflie_demo" type="traj_executor.py" output="screen" args="id 1" /> -->
	<!-- <node name="traj_executor2" pkg="crazyflie_demo" type="traj_executor.py" output="screen" args="id 2" /> -->

</launch>
